hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueHandler.java

```diff
@@ -1522,7 +1549,6 @@ private void deleteInternal(Container container, boolean force)
     }
     // Avoid holding write locks for disk operations
     container.delete();
-    sendICR(container);
```

Comment:
I would just move the ICR sending immediately before `container.delete()`:
```
sendICR(container);
container.delete();
```

This puts the ICR after the move to the tmp dir which is fast, but before the slower filesystem `rm`. The current implementation puts the ICR before moving to tmp which will cause problems if the move fails. SCM will not retry the delete after getting the ICR so it may at a later point try to re-import the container to this volume which will fail since the directory already exists.

Reply:
I am happy to move it there, but the `removeContainer` call also removes the rocksDB entries, which could be slow if the disks are overloaded I guess. Probably depends on how many blocks are in the container and how much data it must read and remove from RocksDB.